tests/fips/openjdk/openjdk_fips.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openjdk fips tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: openjdk expect
+#
+# Summary: FIPS: openjdk
+#          Jira feature: SLE-21206
+#          FIPS 140-3: make OpenJDK be able to use the NSS certified crypto
+#
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils qw(is_sle);
+
+sub run {
+    my $self = @_;
+
+    my $interactive_str = [
+        {
+            prompt => qr/Enter new password/m,
+            key => 'ret',
+        },
+        {
+            prompt => qr/Re-enter password/m,
+            key => 'ret',
+        },
+    ];
+
+    select_console 'root-console';
+    # Install mozilla-nss-tools
+    zypper_call('in mozilla-nss-tools');
+
+    # configure nssdb
+    assert_script_run("mkdir /etc/pki/nssdb");
+    script_run_interactive("certutil -d /etc/pki/nssdb -N", $interactive_str, 30);
+    assert_script_run("chmod og+r /etc/pki/nssdb/*");
+
+    # Install openJDK
+    zypper_call('in java-11-openjdk');
+    zypper_call('in java-11-openjdk-devel');
+
+    # Simple Java Crypto Test
+    zypper_call('in git-core');
+    assert_script_run("cd ~;git clone -q https://github.com/ecki/JavaCryptoTest");
+    script_run("cd ~/JavaCryptoTest/src/main/java/");
+    script_run("javac net/eckenfels/test/jce/JCEProviderInfo.java");
+    my $crypto = script_output("java -cp ~/JavaCryptoTest/src/main/java/ net.eckenfels.test.jce.JCEProviderInfo");
+    record_info("Cannot list all crypto providers", result => 'fail') if ($crypto !~ /Listing all JCA Security Providers/);
+    #Prepare testing data
+    my $JDK_TCHECK = get_var("JDK_TCHECK", "https://gitlab.suse.de/QA-APAC-I/testing/-/raw/master/data/openjdk/Tcheck.java");
+    assert_script_run("cd ~;wget --quiet --no-check-certificate $JDK_TCHECK");
+    assert_script_run("chmod 777 Tcheck.java");
+    assert_script_run("javac Tcheck.java");
+    assert_script_run("java Tcheck > result.txt");
+    my $EX_TCHECK = get_var("EX_TCHECK", "https://gitlab.suse.de/QA-APAC-I/testing/-/raw/master/data/openjdk/Tcheck.txt");
+    assert_script_run("wget --quiet --no-check-certificate $EX_TCHECK");
+    script_run("ls");
```

Comment:
If you want to list the file, I would suggest you record info there, anyway, depend on you

Reply:
I just used this to debug. Had deleted